keps/sig-node/4753-expose-ownerref-downard-api/README.md

```diff
@@ -0,0 +1,592 @@
+- [ ] (R) Production readiness review approved
+- [ ] "Implementation History" section is up-to-date for milestone
+- [ ] User-facing documentation has been created in [kubernetes/website], for publication to [kubernetes.io]
+- [ ] Supporting documentation—e.g., additional design documents, links to mailing list discussions/SIG meetings, relevant PRs/issues, release notes
+
+<!--
+**Note:** This checklist is iterative and should be reviewed and updated every time this enhancement is being considered for a milestone.
+-->
+
+[kubernetes.io]: https://kubernetes.io/
+[kubernetes/enhancements]: https://git.k8s.io/enhancements
+[kubernetes/kubernetes]: https://git.k8s.io/kubernetes
+[kubernetes/website]: https://git.k8s.io/website
+
+## Summary
+
+Today when a pod wants to pass its 'onwerReferences' to a new object it manage (a ConfigMap, a Secret, etc), it needs to do it call the API server to get it's own ownerReferences and then pass it to the new object. 
+The pod then needs GET access on all the intermediate resources which it may not have, and giving it that access is a security risk as it can access other resources it should not have access to.
+This KEP proposes to expose the ownerReferences via `valueFrom` and downward API, so that the pod can directly pass its `ownerReferences` to the managed/created new object without having to call the API server.
+
+This KEP is intentionally scoped small, and intends to keep the scope just to `ownerReferences` and not to other fields of the owner object to prevent security risks.
+
+## Motivation
+
+Deployments and ReplicaSets can orphan and adopt pods dinamically. Any additional information we provide can change over time. On day 1 a pod named "foobar-deadbeef93-xyz76" is owned by replicaset "foobar-deadbeef93" which is owned by deployment "foobar", but after a node restart, the pod name could change to "foobar-deadbeef93-abc12", leaving the object created by "foobar-deadbeef93-xyz76" orphan, and triggering unwanted behavior. Enabling the pod to pass its ownerReferences to the new object it manages will prevent objects from being orphaned by pods that are owned by a higher level object.
+
+### Goals
+
+* Gain access to pod `ownerReferences` on pods through volume mounts.
+* Gain access to pod `ownerReferences` on pods through environmental variables.
+
+### Non-Goals
+
+- Not to expose additional pod info outside of `metadata.ownerReferences`.
+
+## Proposal
+
+The initial design includes:
+
+* Being able to pass pod `ownerReferences` to volumes with fieldPath of `metadata.ownerReferences`.
+* Being able to pass pod `ownerReferences` to environmental variables with fieldPath of `metadata.ownerReferences`.
+
+### User Stories (Optional)
+
+#### Story 1
+xref https://github.com/kubernetes-sigs/node-feature-discovery/issues/1752
+
+CustomResource object created by a pod owned by a DaemonSet is garbage collected when the pod is deleted by unexpected reasons. This triggers unwanted behavior in the system.
```

Comment:
Ahh, the DaemonSet part helps clarify a little.  Can you make this even more concrete?